sled-agent/src/services.rs

```diff
@@ -1618,18 +1618,82 @@ impl ServiceManager {
                 zone:
                     OmicronZoneConfig {
                         zone_type: OmicronZoneType::ClickhouseServer { .. },
-                        underlay_address: _,
+                        underlay_address,
                         ..
                     },
                 ..
             }) => {
-                // We aren't yet deploying this service
-                error!(
-                    &self.inner.log,
-                    "Deploying ClickhouseServer zones is not yet supported"
-                );
+                let Some(info) = self.inner.sled_info.get() else {
+                    return Err(Error::SledAgentNotReady);
+                };
+
+                let listen_addr = *underlay_address;
+                let listen_port = CLICKHOUSE_PORT.to_string();
+
+                let nw_setup_service = Self::zone_network_setup_install(
+                    Some(&info.underlay_address),
+                    &installed_zone,
+                    &[listen_addr],
+                )?;
+
+                let dns_service = Self::dns_install(info, None, &None).await?;
+
+                let config = PropertyGroupBuilder::new("config")
+                    .add_property(
+                        "listen_addr",
+                        "astring",
+                        listen_addr.to_string(),
+                    )
+                    .add_property("listen_port", "astring", listen_port)
+                    .add_property("store", "astring", "/data");
```

Comment:
This is fine for testing, but this is not how I anticipate the system working.  When the zone is launched, a clickhouse server will **not** be running. The `clickhouse-admin` dropshot server instead will be waiting for a [`ReplicaConfig`](https://github.com/oxidecomputer/clickward/blob/main/src/config.rs#L12) to be pushed to it so that it can generate a config file and start the service. In the case of an existing dataset, the config file may already be there, along with a cached version of the `ReplicaConfig` input and a generation number, so that only updates generate new config files. Once there is a valid config file, then `clickhouse-admin` will start the server. 

The only properties passed through should be those needed for starting the `clickhouse-admin` server. This allows us to use a self-assembling zone and not go through sled-agent when the configuration changes.

Reply:
ah sweet! Should I leave this code as is for now until we've modified clickhouse-admin to do what you mention? I'd like to leave replicated clickhouse working with this PR even if we remove things later

Reply:
> ah sweet! Should I leave this code as is for now until we've modified clickhouse-admin to do what you mention? I'd like to leave replicated clickhouse working with this PR even if we remove things later

Absolutely leave it for now. Sorry again for the confusion. Everything in this PR works, so no reason to break it for some future code! Thanks again for hanging with me.